android/src/main/java/com/appfolio/uploader/UploaderModule.kt

```diff
@@ -325,25 +325,37 @@ class UploaderModule(val reactContext: ReactApplicationContext) : ReactContextBa
     val uploads = Arguments.createArray()
 
     for (info in workInfos) {
+      if (info.state === WorkInfo.State.SUCCEEDED || info.state === WorkInfo.State.FAILED) {
+        continue
+      }
+
       val upload = Arguments.createMap()
       
       val idTag = info.tags.toTypedArray().find { it.startsWith(UploadWorker::class.java.simpleName) }
       upload.putString("id", idTag?.removePrefix("${UploadWorker::class.java.simpleName}-"))
 
       when(info.state) {
-        WorkInfo.State.SUCCEEDED, WorkInfo.State.FAILED ->
-          upload.putString("state", "completed")
         WorkInfo.State.RUNNING ->
           upload.putString("state", "running")
```

Comment:
I was confused by adding `running` in two places and possible duplicates. Adding parts of my convo with Karthik in case it helps anyone else understand what's going on here:


> WorkManager is unaware of what the "work" is - we define what it is. Our "work" definition is to simply start an upload in a thread - so when the thread starts, workmanager's work is done.
> this is why our assumption of "success" state was causing issues -- since we incorrectly assumed uploads were complete when WM says "success"

> 
> WorkManager’s running  state relates to spinning up a thread. A thread which is responsible for adding tasks to the UploadManager.taskList. Once this thread is actually going WorkManager is in success state.
> Therefore, for an upload to be in the taskList it’s WM state must be success
> so it’s therefore impossible to be in the tasklist and in the  WM running state